packages/amplify-category-auth/index.js

```diff
@@ -120,28 +120,49 @@ async function externalAuthEnable(context, externalCategory, resourceName, requi
   };
 
   try {
-    authProps = removeDeprecatedProps(authProps);
+    authProps = await removeDeprecatedProps(authProps);
     await copyCfnTemplate(context, category, authProps, cfnFilename);
-    saveResourceParameters(context, provider, category, authProps.resourceName, authProps, ENV_SPECIFIC_PARAMS);
+    await saveResourceParameters(context, provider, category, authProps.resourceName, authProps, ENV_SPECIFIC_PARAMS);
+    const resourceDirPath = path.join(amplify.pathManager.getBackendDirPath(), 'auth', authProps.resourceName, 'parameters.json');
+    const authParameters = await amplify.readJsonFile(resourceDirPath);
     if (!authExists) {
       const options = {
         service: 'Cognito',
         providerPlugin: 'awscloudformation',
       };
-      const resourceDirPath = path.join(amplify.pathManager.getBackendDirPath(), '/auth/', authProps.resourceName, 'parameters.json');
-      const authParameters = amplify.readJsonFile(resourceDirPath);
 
       if (authParameters.dependsOn) {
         options.dependsOn = authParameters.dependsOn;
       }
       await amplify.updateamplifyMetaAfterResourceAdd(category, authProps.resourceName, options);
     }
+
+    // Update Identity Pool dependency attributes on userpool groups
+    const allResources = context.amplify.getProjectMeta();
+    if (allResources.auth && allResources.auth.userPoolGroups) {
+      let attributesm = ['UserPoolId', 'AppClientIDWeb', 'AppClientID'];
+      if (authParameters.identityPoolName) {
+        attributes.push('IdentityPoolId');
+      }
+      const userPoolGroupDependsOn = [
+        {
+          category: 'auth',
+          resourceName: authProps.resourceName,
+          attributes,
+        },
+      ];
+
+      amplify.updateamplifyMetaAfterResourceUpdate('auth', 'userPoolGroups', 'dependsOn', userPoolGroupDependsOn);
```

Comment:
This code almost identical to that of what we have in https://github.com/aws-amplify/amplify-cli/blob/144d69785a8ff8a2a83fefe47b6f195861f51245/packages/amplify-category-auth/commands/auth/update.js#L74-L91. Can we just move it to a method instead of having repeated code

Reply:
Same comment on above:
attilah 6 days ago Member
Same as above, can't we share this logic as well and just have it once in the codebase?

 @kaustavghosh06
kaustavghosh06 4 days ago Author Member
Yeah, as I was diving through the code there's a refactor which I'm planning to basically consolidate the enable/update/externalAuth functions since it has a bunch of repeated code. It'll be a part of another PR.

Reply:
Dont you think it would easier to move it a separate function now as the duplicate code is introduced now. We could always do refactoring, but moving it out now would make one less thing to worry about. I will leave it up to you to decide how you want to handle this